File: tests/testdata/codegen/using-object-as-input-for-any-pp/go/using-object-as-input-for-any.go
package main

import (
	"github.com/pulumi/pulumi-aws-native/sdk/go/aws/iam"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := iam.NewRole(ctx, "role", &iam.RoleArgs{
			RoleName: "ScriptIAMRole",
			AssumeRolePolicyDocument: pulumi.Any(map[string]interface{}{
				"Version": "2012-10-17",
				"Statement": []map[string]interface{}{
					map[string]interface{}{
						"Effect": "Allow",
						"Action": "sts:AssumeRole",
						"Principal": map[string]interface{}{
							"Service": []string{
								"cloudformation.amazonaws.com",
								"gamelift.amazonaws.com",
							},
						},
					},
				},
			}),
		})
		if err != nil {
			return err
		}
		return nil
	})
}

File: tests/testdata/codegen/using-object-as-input-for-any-pp/nodejs/using-object-as-input-for-any.ts
import * as pulumi from "@pulumi/pulumi";
import * as aws_native from "@pulumi/aws-native";

const role = new aws_native.iam.Role("role", {
    roleName: "ScriptIAMRole",
    assumeRolePolicyDocument: {
        Version: "2012-10-17",
        Statement: [{
            Effect: "Allow",
            Action: "sts:AssumeRole",
            Principal: {
                Service: [
                    "cloudformation.amazonaws.com",
                    "gamelift.amazonaws.com",
                ],
            },
        }],
    },
});
